### PR TITLE
feat: add support to fetch connection from connection string

### PR DESCRIPTION
**Pull request by adityathebe**

### CODE REVIEW

File: types/envvar.go [hunk 1]
```diff
@@ -14,8 +14,9 @@ const EnvVarType = "env_var"
 // +kubebuilder:object:generate=true
 type EnvVar struct {
 	Name        string        `json:"name,omitempty" yaml:"name,omitempty" protobuf:"bytes,1,opt,name=name"`
-	ValueStatic string        `json:"value,omitempty" yaml:"value,omitempty" protobuf:"bytes,2,opt,name=value"`
-	ValueFrom   *EnvVarSource `json:"valueFrom,omitempty" yaml:"valueFrom,omitempty" protobuf:"bytes,3,opt,name=valueFrom"`
+	Connection  string        `json:"connection,omitempty" yaml:"connection,omitempty" protobuf:"bytes,2,opt,name=connection"`
```
Comment (adityathebe) [inline types/envvar.go:17]: Are these protobuf struct tags supposed to be written manually?

File: connection.go [hunk 2]
```diff
@@ -10,6 +11,40 @@ import (
...
+	}
+
+	if parts[0] == "" || parts[1] == "" {
+		return
+	}
+
+	return parts[1], parts[0], true
+}
+
+func FindConnectionFromConnectionString(ctx context.Context, db *gorm.DB, connectionString string) (*models.Connection, error) {
```
Comment (moshloop) [inline connection.go:34]:
```suggestion
func FindConnectionByURL(ctx context.Context, db *gorm.DB, url string) (*models.Connection, error) {
```

File: connection.go [hunk 3]
```diff
@@ -10,21 +12,86 @@ import (
...
+	if parts[0] == "" || parts[1] == "" {
+		return
+	}
+
+	return parts[1], parts[0], true
+}
+
+// FindHydratedConnectionByURL retrieves a connection from the given connection string.
+// The connection string is expected to be of the form: connection://<type>/<name>
+func FindHydratedConnectionByURL(ctx context.Context, db *gorm.DB, k8sClient kubernetes.Interface, namespace, connectionString string) (*models.Connection, error) {
```
Comment (moshloop) [inline connection.go:43]:
```suggestion
func HydratedConnectionByURL(ctx context.Context, db *gorm.DB, k8sClient kubernetes.Interface, namespace, connectionString string) (*models.Connection, error) {
```

File: connection.go [hunk 4]
```diff
@@ -10,21 +12,90 @@ import (
...
 func GetConnection(ctx context.Context, client kubernetes.Interface, db *gorm.DB, connectionType string, name string, namespace string) (*models.Connection, error) {
 	connection, err := FindConnection(ctx, db, connectionType, name)
 	if err != nil {
 		return nil, err
 	}
-	return HydrateConnection(ctx, client, db, connection, namespace)
 
+	if connection == nil {
+		return nil, ErrNotFound
+	}
```
Comment (adityathebe) [inline connection.go:96]: @moshloop we could do this or I could rename the func to `HydratedConnection` like `HydratedConnectionByURL`

**Pull request merged**